core/unit_test/TestComplex.hpp

```diff
@@ -519,9 +519,13 @@ TEST(TEST_CATEGORY, complex_operations_arithmetic_types_overloads) {
   ASSERT_EQ(Kokkos::conj(1), Kokkos::complex<double>(1));
   ASSERT_EQ(Kokkos::conj(2.f), Kokkos::complex<float>(2.f));
   ASSERT_EQ(Kokkos::conj(3.), Kokkos::complex<double>(3.));
+// long double has size 12 but Kokkos::complex requires 2*sizeof(T) to be a
+// power of two.
```

Comment:
I don't get what this has to do with 32bit compilation?

Reply:
I don't know either but I was seeing 
```
In file included from /__w/kokkos/kokkos/core/src/Kokkos_MathematicalSpecialFunctions.hpp:31,
                 from /__w/kokkos/kokkos/core/src/Kokkos_Core.hpp:54,
                 from /__w/kokkos/kokkos/core/unit_test/TestComplex.hpp:17,
                 from /__w/kokkos/kokkos/builddir/core/unit_test/openmp/TestOpenMP_Complex.cpp:2:
/__w/kokkos/kokkos/core/src/Kokkos_Complex.hpp: In instantiation of 'class Kokkos::complex<long double>':
/__w/kokkos/kokkos/core/unit_test/TestComplex.hpp:522:3:   required from here
/__w/kokkos/kokkos/core/src/Kokkos_Complex.hpp:46:9: error: requested alignment '24' is not a positive power of 2
   46 |         complex {
      |         ^~~~~~~
```
in https://github.com/masterleinad/kokkos/actions/runs/4430717022/jobs/7772798043.

Reply:
I mean, we could rather test for `sizeof(long double)`.